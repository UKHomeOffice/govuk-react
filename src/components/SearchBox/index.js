// https://govuk-static.herokuapp.com/component-guide/search

import React from "react";
import PropTypes from "prop-types";
import glamorous from "glamorous";

import { LIGHT_BLUE, WHITE, YELLOW } from "govuk-colours";

const SearchBoxWrapper = glamorous.div({
  boxSizing: "border-box",
  display: "flex",
  width: "100%",
  background: WHITE
});

// css normalize is hiding the input:search clear SearchButton
const InputSearchBox = glamorous.input({
  width: "100%",
  height: "40px",
  padding: "6px",
  margin: 0,
  border: "solid 1px #bfc1c3",
  borderRight: 0,
  boxSizing: "border-box",
  fontFamily: "'nta', Arial, sans-serif",
  fontWeight: 400,
  textTransform: "none",
  fontSize: "16px",
  lineHeight: "1.75",
  background: "#fff",
  borderRadius: 0,
  WebkitAppearance: "none",
  ":focus": {
    marginRight: "3px",
    outline: `3px solid ${YELLOW}`,
    outlineOffset: 0,
    " ~ button": {
      width: "46px"
    }
  }
});

const SearchButton = glamorous.button({
  backgroundColor: LIGHT_BLUE,
  border: 0,
  display: "block",
  color: "#fff",
  position: "relative",
  padding: "0",
  width: "45px",
  height: "40px",
  backgroundImage:
    "url(https://govuk-static.herokuapp.com/static/search-button-ca89b2a79f944909ceb7370d3f0b78811d32b96e883348fcd8886f63dd619585.png)",
  backgroundRepeat: "no-repeat",
  backgroundPosition: "2px 50%",
  textIndent: "-5000px",
  overflow: "hidden",
  ":focus": {
    outline: `3px solid ${YELLOW}`,
    outlineOffset: 0
  }
});

const SearchBox = ({ placeholder }) => (
  <SearchBoxWrapper>
    <InputSearchBox type="search" placeholder={placeholder} />
    <SearchButton>Search</SearchButton>
  </SearchBoxWrapper>
);

SearchBox.defaultProps = {
  placeholder: undefined
};

SearchBox.propTypes = {
  placeholder: PropTypes.string
};

export default SearchBox;
