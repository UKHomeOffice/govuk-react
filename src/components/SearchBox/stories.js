import React from "react";
import { storiesOf } from "@storybook/react";

import Layout from "../Layout/index";
import GridRow from "../GridRow/index";
import GridCol from "../GridCol/index";

import SearchBox from ".";

storiesOf("SearchBox", module).add("SearchBox", () => (
  <Layout>
    <GridRow>
      <GridCol>
        <SearchBox placeholder="Search GOV.UK">SearchBox example</SearchBox>
      </GridCol>
    </GridRow>
  </Layout>
));
