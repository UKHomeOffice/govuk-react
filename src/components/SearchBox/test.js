import React from "react";
import ReactDOM from "react-dom";
import SearchBox from "./";

describe(SearchBox, () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<SearchBox>Example</SearchBox>, div);
  });
});
