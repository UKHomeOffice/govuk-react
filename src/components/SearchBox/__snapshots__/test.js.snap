// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox matches wrapper snapshot: wrapper mount 1`] = `
.glamor-2,
[data-glamor-2] {
  box-sizing: border-box;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  width: 100%;
  background: #ffffff;
}

.glamor-0,
[data-glamor-0] {
  width: 100%;
  height: 40px;
  padding: 6px;
  margin: 0;
  border: solid 1px #bfc1c3;
  border-right: 0;
  box-sizing: border-box;
  font-family: 'nta', Arial, sans-serif;
  font-weight: 400;
  text-transform: none;
  font-size: 16px;
  line-height: 1.75;
  background: #fff;
  border-radius: 0;
  -webkit-appearance: none;
}

.glamor-0:focus,
[data-glamor-0]:focus {
  margin-right: 3px;
  outline: 3px solid #ffbf47;
  outline-offset: 0;
}

.glamor-0:focus ~ button,
[data-glamor-0]:focus ~ button {
  width: 46px;
}

.glamor-1,
[data-glamor-1] {
  background-color: #2b8cc4;
  border: 0;
  display: block;
  color: #fff;
  position: relative;
  padding: 0px;
  width: 45px;
  height: 40px;
  background-image: url(https://govuk-static.herokuapp.com/static/search-button-ca89b2a79f944909ceb7370d3f0b78811d32b96e883348fcd8886f63dd619585.png);
  background-repeat: no-repeat;
  background-position: 2px 50%;
  text-indent: -5000px;
  overflow: hidden;
}

.glamor-1:focus,
[data-glamor-1]:focus {
  outline: 3px solid #ffbf47;
  outline-offset: 0;
}

<SearchBox
  placeholder="example"
>
  <glamorous(div)>
    <div
      className="glamor-2"
    >
      <glamorous(input)
        placeholder="example"
        type="search"
      >
        <input
          className="glamor-0"
          placeholder="example"
          type="search"
        />
      </glamorous(input)>
      <glamorous(button)>
        <button
          className="glamor-1"
        >
          Search
        </button>
      </glamorous(button)>
    </div>
  </glamorous(div)>
</SearchBox>
`;
